providers/legacyprov.c

```diff
@@ -184,11 +184,34 @@ int OSSL_provider_init(const OSSL_PROVIDER *provider,
 
     *out = legacy_dispatch_table;
 
+#if 1
     /*
      * We want to make sure that all calls from this provider that requires
      * a library context use the same context as the one used to call our
      * functions.  We do that by passing it along as the provider context.
      */
-    *provctx = c_get_libctx(provider);
+
+    /*
+     * TODO(3.0): FIXME FIXME FIXME DANGER DANGER DANGER
+     * 
+     * We should really replace this with the code below - but for some reason
+     * I'm getting a linker error when I do so.
+     */ 
+    *provctx = c_get_libctx(handle);
+#else
+    {
+        OPENSSL_CTX *libctx = OPENSSL_CTX_new();
+
+        if (libctx == NULL)
+            return 0;
+
+        /*
+         * We would really like to use the same libctx that the application uses
+         * - but since the application may be linked against a different version
+         * of libcrypto to us we cannot.
+         */
+        *provctx = libctx;
+    }
+#endif
```

Comment:
Conidering I've done the latter in #11803, I assume that this particular change is simply going to disappear

Reply:
Correct